scripts/lib/CIME/BuildTools/cmakemacroswriter.py

```diff
@@ -117,3 +118,42 @@ def end_ifeq(self):
         """
         self.indent_left()
         self.write_line("endif()")
+
+    def _transform_value(self, name, value):
+        """Some elements need their values transformed in some way for CMake to handle them properly.
+        This method does those transformations.
+
+        Args:
+        - name (str): name of element
+        - value (str): value of element
+
+        Returns transformed value
+        """
+
+        value_transformed = value
+        if self._element_needs_whitespace_removal(name):
+            value_transformed = value_transformed.strip()
+
+        return value_transformed
+
+    def _element_needs_whitespace_removal(self, name):
```

Comment:
I wondered about that. I thought there might be some variables where there was deliberately some leading or trailing white space so that when they are concatenated together you get something like `-foo -bar` rather than `-foo-bar`. But I didn't look closely or experiment with this. If you think it's safe to always remove whitespace, then I can give it a shot on the four machines that now have unit testing support.